BankPrototype/deal/src/test/java/com/bankprototype/deal/web/controller/DealControllerTest.java

```diff
@@ -0,0 +1,274 @@
+                .passportIssueDate(LocalDate.of(2020, 07, 07))
+                .passportIssueBranch("issue Branch")
+                .maritalStatus(MaritalStatus.SINGLE)
+                .dependentAmount(BigDecimal.valueOf(1159689))
+                .employment(employmentDTO)
+                .account("testAccount")
+                .isInsuranceEnabled(true)
+                .isSalaryClient(true)
+                .build();
+
+        StatusHistory applicationStatusHistory = StatusHistory.builder()
+                .status(ApplicationStatus.PREAPPROVAL)
+                .time(LocalDateTime.now())
+                .changeType(ChangeType.AUTOMATIC)
+                .build();
+        List<StatusHistory> listStatus = List.of(applicationStatusHistory);
+
+        Application applicationTest = Application.builder()
+                .applicationId(1L)
+                .applicationId(1L).build();
+
+        Client clientTest = random(Client.class);
+        clientTest.setClientId(1L);
+
+        Credit credit = new Credit();
+
+        when(applicationService.getApplicationById(any()))
+                .thenReturn(applicationTest);
+
+        when(clientService.updateClient(any(),any()))
+                .thenReturn(clientTest);
+
+        when(creditService.createScoringDataDTO(any(),any(),any()))
+                .thenReturn(scoringDataDTO);
+
+        when(creditService.createCredit(any(),any()))
+                .thenReturn(credit);
+
+        ResultActions response = mockMvc.perform(post("/deal/calculate/1")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("{\n" +
+                                "  \"gender\": \"MALE\",\n" +
+                                "  \"maritalStatus\": \"SINGLE\",\n" +
+                                "  \"dependentAmount\": 1000000,\n" +
+                                "  \"passportIssueDate\": \"2020-07-30\",\n" +
+                                "  \"passportIssueBrach\": \"passportIssue\",\n" +
+                                "  \"employment\": {\n" +
+                                "    \"employmentStatus\": \"EMPLOYED\",\n" +
+                                "    \"employerINN\": \"string\",\n" +
+                                "    \"salary\": 100000,\n" +
+                                "    \"position\": \"MIDDLE_MANAGER\",\n" +
+                                "    \"workExperienceTotal\": 48,\n" +
+                                "    \"workExperienceCurrent\": 24\n" +
+                                "  },\n" +
+                                "  \"account\": \"account\"\n" +
+                                "}")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(MockMvcResultMatchers.status().isOk());
+
+        System.out.println(response);
+    }
+
+    @Test
+    void completionRegistrationAndCalculateFullCredit_ExceptionResourceNotFound() throws Exception {
```

Comment:
Тест упал
java.lang.NullPointerException: Cannot invoke "com.bankprototype.deal.repository.dao.Client.getClientId()" because the return value of "com.bankprototype.deal.repository.dao.Application.getClientId()" is null
Не задан clientId в объекте application